### PR TITLE
Feat feed functionality

### PR DESCRIPTION
**Pull request by coldpigli**

### CODE REVIEW

File: src/components/Feedlisting/Feedlisting.jsx [hunk 1]
```diff
@@ -9,6 +9,8 @@ const Feedlisting = () => {
   const {userData} = useSelector((state)=>state.auth);
   const {postList} = useSelector((state)=>state.posts);
   const dispatch = useDispatch();
+  console.log("User data", userData);
+  console.log("Post list", postList);
```
Comment (vaibhav-mahalle) [inline src/components/Feedlisting/Feedlisting.jsx]: is this console.log() for future reference, if not remove it.

File: src/components/InfoBar/InfoBar.jsx [hunk 2]
```diff
@@ -4,14 +4,14 @@ import {BiChevronRight} from 'react-icons/bi'
...
+        <Box p='0.5rem' borderRadius='1rem' bg='#242731'>
           <HStack justify='space-between'>
             <HStack>
-              <Box bg='#FFE7CA' p='0.5rem' borderRadius='1rem'>
+              <Box bg='#6C5DD3' p='0.5rem' borderRadius='1rem'>
                 <Icon w={8} h={8} as={SiCloudfoundry}/>
               </Box>
               <VStack align='stretch' spacing='0'>
-                <Text fontSize='sm'>Thoughts Shared</Text>
+                <Text fontSize='sm' color="#808191">Followers</Text>
```
Comment (vaibhav-mahalle) [inline src/components/InfoBar/InfoBar.jsx:14]: you could use color variable.

File: src/components/CreatePost/CreatePost.jsx [hunk 3]
```diff
@@ -1,46 +1,98 @@
...
-            placeholder="Pen your thoughts down"
-            variant="unstyled"
-            bg="#242731"
-            focusBorderColor="none"
-            resize="none"
-            border="none"
-          />
+          <FormControl>
+            <Textarea
+              isInvalid={formik.errors.postData}
```
Comment (vaibhav-mahalle) [inline src/components/CreatePost/CreatePost.jsx:46]: Liked the use of formik here.

**Review by vaibhav-mahalle — APPROVED** [review]
suggested some minor changes please look into it.

File: src/components/PostItem/PostItem.jsx [hunk 4]
```diff
@@ -48,36 +72,24 @@ const PostItem = ({post}) => {
...
-            colorScheme="black"
-            fontSize="1.5rem"
-            icon={<MdBookmarkBorder />}
-            _focus={{
-              outline: "none",
-            }}
-          />
+        <HStack justify="space-between">
+          {engageButtons.map((item) => (
+            <HStack color="#808191" align="center">
```
Comment (nishasen) [inline src/components/PostItem/PostItem.jsx:81]: use variable colors

File: src/services/postServices.js [hunk 5]
```diff
@@ -5,13 +5,27 @@ const getAllPosts = createAsyncThunk("posts/getAllPosts",
...
     }
+});
+
+const addNewPost = createAsyncThunk("posts/addNewPost",
+async(postData, {getState, rejectWithValue})=>{
+    try{
+        const {authToken} = getState().auth;
+        console.log("PostData", postData);
+        const res = await axios.post('/api/posts',{postData},{headers: {authorization: authToken}});
+        console.log("Returned data after adding your post",res.data)
```
Comment (nishasen) [inline src/services/postServices.js:20]: remove the console logs

**Review by nishasen — APPROVED** [review]
Everything looks good, have added a few suggestions have a look into it

File: src/components/PostItem/PostItem.jsx (hunk 4 above)
Comment (coldpigli) [reply src/components/PostItem/PostItem.jsx:81]: Yes, will do it in the next PR.

File: src/components/InfoBar/InfoBar.jsx (hunk 2 above)
Comment (coldpigli) [reply src/components/InfoBar/InfoBar.jsx:14]: Will definitely do it in the next PR

**Pull request merged**